File: dart/rna-transcription/lib/rna_transcription.dart
class RnaTranscription {
  final Map<String, String> _mappings = {
    'G': 'C',
    'C': 'G',
    'T': 'A',
    'A': 'U',
  };

  String toRna(String dna) => dna.split("").map((e) => _mappings[e]).join("");
}

File: csharp/rna-transcription/RnaTranscription.cs
public static class RnaTranscription
{
    public static string ToRna(string strand) => string.Join("", strand.Select(ch =>
                                                      {
                                                          return ch switch
                                                          {
                                                              'G' => 'C',
                                                              'C' => 'G',
                                                              'T' => 'A',
                                                              'A' => 'U',
                                                              _ => ' '
                                                          };
                                                      }));
}